packages/integration-tests/tests/test-metrics.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+const sinon = require('sinon');
+const test = require('ava');
+
+const { getAggregateMetricQuery, getInvocationCount } = require('../metrics');
+
+test('getAggregateMetricQuery returns the max average for an Average metrics query', async (t) => {
+  const cloudwatch = () => ({
+    getMetricStatistics: (_val) => ({
+      promise: async () => ({ Datapoints: [{ Average: 10 }, { Average: 20 }] }),
+    }),
+  });
+  const actual = await getAggregateMetricQuery({ Statistics: ['Average'] }, cloudwatch);
+  t.is(actual, 20);
+});
+
+test('getAggregateMetricQuery returns the maximum for a Maximum metrics query', async (t) => {
+  const cloudwatch = () => ({
+    getMetricStatistics: (_val) => ({
+      promise: async () => ({ Datapoints: [{ Maximum: 10 }, { Maximum: 30 }] }),
+    }),
+  });
+  const actual = await getAggregateMetricQuery({ Statistics: ['Maximum'] }, cloudwatch);
+  t.is(actual, 30);
+});
+
+test('getAggregateMetricQuery returns the minimum for a Minimum metrics query', async (t) => {
+  const cloudwatch = () => ({
+    getMetricStatistics: (_val) => ({
+      promise: async () => ({ Datapoints: [{ Minimum: 10 }, { Minimum: 30 }] }),
+    }),
+  });
+  const actual = await getAggregateMetricQuery({ Statistics: ['Minimum'] }, cloudwatch);
+  t.is(actual, 10);
+});
+
+test('getAggregateMetricQuery returns the total for a Sum query', async (t) => {
+  const cloudwatch = () => ({
+    getMetricStatistics: (_val) => ({
+      promise: async () => ({ Datapoints: [{ Sum: 10 }, { Sum: 30 }] }),
+    }),
+  });
+  const actual = await getAggregateMetricQuery({ Statistics: ['Sum'] }, cloudwatch);
+  t.is(actual, 40);
+});
+
+test('getInvocationCount returns the expected invocation count', async (t) => {
+  const aggregateMetricQueryFunction = sinon.stub();
+  aggregateMetricQueryFunction.returns(5);
+  const actual = await getInvocationCount({
+    aggregateMetricQueryFunction,
+    beginTime: new Date(),
+    interval: 15,
+    lambda: 'fakeLambda',
+    maxCount: 8,
+    minCount: 3,
+    timeout: 150,
+  });
+  t.is(actual, 5);
+});
+
+test('getInvocationCount returns immediately on if result is greater than the max count', async (t) => {
```

Comment:
Should test string be:
```suggestion
test('getInvocationCount returns immediately if result is greater than the max count', async (t) => {
```
? Or am I misunderstanding something... 🤔